### PR TITLE
added loss scale

### PR DESCRIPTION
**Pull request by Atticus1806**
Added loss scale parameter to the `mark_as_loss` function in a similiar style to how it already sets `loss = as_is` . Cf. #9

### CODE REVIEW

File: models/base.py [hunk 1]
```diff
@@ -120,12 +120,15 @@ def __init__(self, maker: ILayerMaker, layer_dict: LayerDictRaw):
     self.maker = maker
     self.layer_dict = layer_dict
 
-  def mark_as_loss(self):
+  def mark_as_loss(self, loss_scale=1.0):
```
Comment (albertz) [inline models/base.py:123]: Better:
```suggestion
  def mark_as_loss(self, loss_scale: Optional[float] = 1.0):
```

File: models/base.py [hunk 2]
```diff
@@ -120,12 +120,15 @@ def __init__(self, maker: ILayerMaker, layer_dict: LayerDictRaw):
...
     """
     Mark this as a loss.
+    :param loss_scale: scale factor for loss (1.0 by default)
```
Comment (albertz) [inline models/base.py:126]: Make an empty line after the description before the first `:param` (consistency).

Or maybe even better: Just remove this line. It doesn't really add any useful information.
```suggestion
```

File: models/base.py [hunk 3]
```diff
@@ -120,12 +120,15 @@ def __init__(self, maker: ILayerMaker, layer_dict: LayerDictRaw):
...
     """
     assert "loss" not in self.layer_dict
     self.layer_dict["loss"] = "as_is"
+    assert "loss_scale" not in self.layer_dict
+    self.layer_dict["loss_scale"] = loss_scale
```
Comment (albertz) [inline models/base.py:131]:
```suggestion
    if loss_scale is not None:
      assert "loss_scale" not in self.layer_dict
      self.layer_dict["loss_scale"] = loss_scale
```

**Review by albertz — APPROVED** [review]
